### ISSUE TITLE
no-extra-parens error when doing a type assertion

### ISSUE THREAD

**Issue by screendriver** [eslint/typescript-eslint-parser#465]
**What version of TypeScript are you using?**

2.8.1

**What version of `typescript-eslint-parser` are you using?**

14.0.0

**What code were you trying to parse?**

```ts
t.true((me.get as SinonStub).calledWithExactly('/foo', other));
```

**What did you expect to happen?**

ESLint should not complain.

**What happened?**

`error |   Gratuitous parentheses around expression     |        no-extra-parens`

**Comment by JamesHenry** [comment]
I think a custom rule may need to be created in the [plugin](https://github.com/nzakas/eslint-plugin-typescript) to handle this. It would be fantastic if one of the people affected by this issue could contribute to it.

**Comment by JamesHenry** [comment]
This issue has been migrated to the new project: typescript-eslint/typescript-eslint#12

Thanks!
